week7/answers.sql

```diff
@@ -1,3 +1,314 @@
+	team_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	name VARCHAR(30)
+);
+
+CREATE TABLE team_members (
+	team_member_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	team_id INT UNSIGNED,
+	character_id INT UNSIGNED,
+  
+	CONSTRAINT team_members_fk_teams
+		FOREIGN KEY (team_id)
+		REFERENCES teams (team_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE,
+  
+	CONSTRAINT team_members_fk_characters
+		FOREIGN KEY (character_id)
+		REFERENCES characters (character_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+CREATE TABLE items (
+	item_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	name VARCHAR(30),
+	armor INT UNSIGNED NOT NULL,
+	damage INT UNSIGNED NOT NULL
+);
+
+CREATE TABLE inventory (
+	inventory_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	character_id INT UNSIGNED,
+	item_id INT UNSIGNED,
+  
+	CONSTRAINT inventory_fk_characters
+		FOREIGN KEY (character_id)
+		REFERENCES characters (character_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE,
+  
+	CONSTRAINT inventory_fk_items
+		FOREIGN KEY (item_id)
+		REFERENCES items (item_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+CREATE TABLE equipped (
+	equipped_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	character_id INT UNSIGNED,
+	item_id INT UNSIGNED,
+  
+	CONSTRAINT equipped_fk_characters
+		FOREIGN KEY (character_id)
+		REFERENCES characters (character_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE,
+  
+	CONSTRAINT equipped_fk_items
+		FOREIGN KEY (item_id)
+		REFERENCES items (item_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+DELIMITER ;;
+CREATE PROCEDURE equip(inventory_id INT UNSIGNED)
+BEGIN
+	DECLARE find_item_by_id INT UNSIGNED;
+	DECLARE id_of_character INT UNSIGNED;
+	DECLARE id_of_item INT UNSIGNED;
+
+	SELECT inventory_id INTO find_item_by_id;
+
+	SELECT 
+		inventory.character_id
+	FROM inventory
+	WHERE inventory.inventory_id = find_item_by_id
+	INTO id_of_character;
+
+	SELECT 
+		inventory.item_id
+	FROM inventory
+	WHERE inventory.inventory_id = find_item_by_id
+	INTO id_of_item;
+
+	INSERT INTO equipped (character_id, item_id) VALUES (id_of_character, id_of_item);
+
+	DELETE FROM inventory WHERE inventory.inventory_id = find_item_by_id; 
+END;;
+
+CREATE PROCEDURE unequip(equipped_id INT UNSIGNED)
+BEGIN
+	DECLARE find_item_by_id INT UNSIGNED;
+	DECLARE id_of_character INT UNSIGNED;
+	DECLARE id_of_item INT UNSIGNED;
+
+	SELECT equipped_id INTO find_item_by_id;
+
+	SELECT 
+		equipped.character_id
+	FROM equipped
+	WHERE equipped.equipped_id = find_item_by_id
+	INTO id_of_character;
+
+	SELECT 
+		equipped.item_id
+	FROM equipped
+	WHERE equipped.equipped_id = find_item_by_id
+	INTO id_of_item;
+
+	INSERT INTO inventory (character_id, item_id) VALUES (id_of_character, id_of_item);
+
+	DELETE FROM equipped WHERE equipped.equipped_id = find_item_by_id; 
+END;;
+
+CREATE PROCEDURE set_winners(team_id INT UNSIGNED)
+BEGIN
+	DECLARE find_winner_by_team_id INT UNSIGNED;
+	SELECT team_id INTO find_winner_by_team_id;
```

Comment:
If you rename `team_id` to something else, you can avoid the additional declaration and copy.